### PR TITLE
Add establish timeouts

### PR DESCRIPTION
**Pull request by marsella**
Adds timeouts to establish functions, using same methods as in #311.

### CODE REVIEW

File: src/bin/merchant/main.rs [hunk 1]
```diff
@@ -58,12 +58,9 @@ pub trait Command {
 #[async_trait]
 impl Command for Run {
     async fn run(self, config: Config) -> Result<(), anyhow::Error> {
-        let database = database(&config)
-            .await
-            .context("Failed to connect to merchant database")?;
```
Comment (gijsvl) [inline src/bin/merchant/main.rs:63]: We lost this error in the refactor

File: src/bin/customer/establish.rs [hunk 2]
```diff
@@ -613,6 +535,247 @@ fn write_establish_json(establishment: &Establishment) -> Result<(), anyhow::Err
     Ok(())
 }
 
+/// Parse the [`Amount`]s passed on the command line into the `Balance` types used throughout.
+fn parse_initial_balances(
```
Comment (gijsvl) [inline src/bin/customer/establish.rs:539]: Easy to unit test

**Review by gijsvl — CHANGES_REQUESTED** [review]
Hard to review because of the refactor, but I've tested the configurations with a low timeout and it seems to correctly handle that.
Just two minor comments.

File: src/bin/merchant/main.rs (hunk 1 above)
Comment (marsella) [reply src/bin/merchant/main.rs:63]: Re-added, thank you.

File: src/bin/customer/establish.rs (hunk 2 above)
Comment (marsella) [reply src/bin/customer/establish.rs:539]: Good call, added a test with the failure cases I know about.

File: src/bin/customer/establish.rs [hunk 3]
```diff
@@ -182,179 +116,89 @@ impl Command for Establish {
...
+            &channel_name,
+            channel_id,
+            database.as_ref(),
+            &config,
+            &merchant_funding_info,
+            &customer_funding_info,
+            &zkabacus_customer_config,
+            self.off_chain,
+        )
+        .await?;
```
Comment (yaymukund) [inline src/bin/customer/establish.rs:146]: I don't feel that separating this code actually makes it more skimmable in practice. At the code level, `fn originate_contract` can do pretty anything that `run()` can do, so I'd want to read them together to have the full picture. 

In fact, I sort of think we should go the other way - I think even the functions that exist solely to add a `with_timeout()` should become inline `async {...}.with_timeout(...)` blocks to keep the protocol logic in one place.

File: src/bin/customer/establish.rs [hunk 4]
```diff
@@ -431,6 +275,81 @@ async fn get_parameters(
...
+    note: Option<Note>,
+) -> Result<(ChannelId, Chan<establish::Initialize>), anyhow::Error> {
+    // Generate randomness for the channel ID
+    let customer_randomness = CustomerRandomness::new(&mut rng);
+
+    // Read the contents of the channel establishment note, if any: this is the justification,
+    // if any is needed, for why the channel should be allowed to be established (format
+    // unspecified, specific to merchant)
+    let note = note
+        .unwrap_or_else(|| zeekoe::customer::cli::Note::String(String::from("")))
```
Comment (yaymukund) [inline src/bin/customer/establish.rs:295]: `unwrap_or_default()`

**Review by yaymukund — CHANGES_REQUESTED** [review]
Left comments inline

File: src/bin/customer/establish.rs (hunk 3 above)
Comment (marsella) [reply src/bin/customer/establish.rs:146]: Yeah, I can get behind this. The sheer volume of parameters plus the non-reusability of the function makes this a worse solution that I was hoping. I will try adding async blocks instead.

File: src/bin/customer/establish.rs [hunk 5]
```diff
@@ -585,6 +507,247 @@ fn write_establish_json(establishment: &Establishment) -> Result<(), anyhow::Err
...
+/// Parse the [`Amount`]s passed on the command line into the `Balance` types used throughout.
+fn parse_initial_balances(
+    deposit: Amount,
+    merchant_deposit: Option<Amount>,
+) -> Result<(CustomerBalance, MerchantBalance), anyhow::Error> {
+    // Format deposit amounts as the correct types
+    let customer_balance = CustomerBalance::try_new(
+        deposit
+            .try_into_minor_units()
+            .ok_or(establish::Error::InvalidDeposit(Customer))?
```
[the remark this answers is not in the document]
Comment (marsella) [reply src/bin/customer/establish.rs]: Yes, good idea. I macro'd these because it is essentially the same for `CustomerBalance` and `MerchantBalance`.

**Review by gijsvl — APPROVED** [review]

File: src/bin/customer/establish.rs (hunk 3 above)
Comment (marsella) [reply src/bin/customer/establish.rs]: This is now done!

**Comment by marsella** [comment]
Moved everything from establish back into `async` blocks. Will wait for re-review or confirmation from @yaymukund before merging.

**Review by yaymukund — APPROVED** [review]
Looks good!

**Pull request merged**